From 53e0a2abe0018b57c67fd398903319d208cb1572 Mon Sep 17 00:00:00 2001
From: Heiko Thiery <heiko.thiery@gmail.com>
Date: Mon, 30 Mar 2020 11:14:18 +0200
Subject: [PATCH] channels: properly set rc to length

It was observed that in case the returned length of ssh_buffer_get_len
was 0 the return value was not properly set to that value. This leads to
timeout failures.

Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>
---
 src/channels.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/channels.c b/src/channels.c
index 5601df80..f9f95666 100644
--- a/src/channels.c
+++ b/src/channels.c
@@ -3118,11 +3118,10 @@ int ssh_channel_poll_timeout(ssh_channel channel, int timeout, int is_stderr)
         goto out;
     }
     len = ssh_buffer_get_len(stdbuf);
+    rc = (int)len;
     if (len > 0) {
         if (len > INT_MAX) {
             rc = SSH_ERROR;
-        } else {
-            rc = (int)len;
         }
         goto out;
     }
-- 
2.20.1

